pkg/ebpf/tracee.go

```diff
@@ -1176,11 +1174,25 @@ func (t *Tracee) attachProbes() error {
 			}
 		}
 
+		// save required probes for later attachment
+		for _, probeDep := range eventDefinition.GetDependencies().GetProbes() {
+			probesToEvents[probeDep] = append(probesToEvents[probeDep], id)
+		}
+	}
+
+	for probe, evtsIDs := range probesToEvents {
 		// attach probes for selected events
-		for _, dep := range eventDefinition.GetDependencies().GetProbes() {
-			err = t.probes.Attach(dep.GetHandle(), t.cgroups)
-			if err != nil && dep.IsRequired() {
-				return errfmt.Errorf("failed to attach required probe: %v", err)
+		err = t.probes.Attach(probe.GetHandle(), t.cgroups)
+		if err != nil && probe.IsRequired() {
+			// if a required probe fails to attach, cancel all events that depend on it
+			for _, evtID := range evtsIDs {
+				evtName := events.Core.GetDefinitionByID(evtID).GetName()
+				logger.Errorw(
+					"Event canceled because of missing probe dependency",
+					"missing probe", probe.GetHandle(), "event", evtName,
+				)
```

Comment:
Sorry for the late comment here @geyslan, but I'm catching up with latest changes before moving on. I believe this change missed "event dependencies", right ? So if a symbol isn't available you are disabling the event directly related to that dependency (all of them) but not the events that depend on that event. 

The direct example, that you're likely solving in the other PR (For the print syscall table), could be:

- HiddenKernelModuleSeeker (depends on symbols "modules", "module_kset", "mod_tree")

So if one of those symbols isn't available, you would disable it directly in the eventsState. But you wouldn't disable the event that depends on it (that could, and probably would) be enabled:

- HiddenKernelModule

Right ? And then there is the fact of the new policy manager (which would also have to have the rules, using those events, disabled from each policy that uses them).

Is that all correct ? And are you addressing all those in the other PR ? Just to accelerate my understanding. Thank you.

Reply:
> So if one of those symbols isn't available, you would disable it directly in the eventsState. But you wouldn't disable the event that depends on it (that could, and probably would) be enabled

After your feedback, I'm making changes to the new PR tackling the case of events dependency.

> And then there is the fact of the new policy manager (which would also have to have the rules, using those events, disabled from each policy that uses them).

Currently, in the event cancellation phase, it would not reach the point of being propagated to BPF filters. So disabling it in policyManager is not necessary now.

In the future, all that logic (eventState/policies) must reside in the policy manager, indeed.